linkedlist.py

```diff
@@ -0,0 +1,118 @@
+# Question 1: Implement a Linked List
+
+class Node:
+    def __init__(self, data=None):
+        self.value = data
+        self.next = None
+
+    def __repr__(self):
+        return f"{self.value} -> {self.next}"
+
+class LinkedList:
+    def __init__(self, head=None):
+        self.head = head
+
+    def __repr__(self):
+        rep = "" + self.head.value
+        while self.head.next:
+            rep += f"-> {self.head.next.value}"
+            self.head = self.head.next
+
+    def __len__(self):
+        c = 0
+        while self.head:
+            c += 1
+            self.head = self.head.next
+        return c
+
+    def __iter__(self):
+        while self.head:
+            yield self.head
+            self.head = self.head.next
+
+    def __contains__(self, data):
+        while self.head:
+            if self.head.value == data:
+                return True
+            self.head = self.head.next
+        return False
+
+    def append(self, data):
```

Comment:
Take another look at this. We can append to the head but then we want to ensure that the new `head` is the newly created node.